### PR TITLE
(feat) add gateway assistant

### PR DESCRIPTION
**Pull request by tphoney**
@dylanratcliffe @DavidS-ovm its probably worth talking about this now, this is a rough estimation based on https://platform.openai.com/docs/api-reference/messages/object
we are only using a string for now.

### CODE REVIEW

File: gateway.proto [hunk 1]
```diff
@@ -205,3 +209,22 @@ message SnapshotLoadResult {
     // a correlation ID to match up requests and responses. this field returns the contents of the request's msgID
     bytes msgID = 4;
 }
+
+message SendChatMessage {
+    // a correlation ID to match up requests and responses. set this to a value unique per connection
+    bytes msgID = 1;
```
Comment (DavidS-ovm) [inline gateway.proto:215]: the `msgId` is only required for request/response pairs like `StoreSnapshot`/`SnapshotStoreResult`. There could be many snapshots be stored at the same time (wasteful, but possible) where some work and some don't. There is only one chat, and message ordering is loosely based on time of dispatch/reception.

File: changes.proto [hunk 2]
```diff
@@ -71,6 +71,8 @@ service ChangesService {
   rpc UpdateChange(UpdateChangeRequest) returns (UpdateChangeResponse);
   // Deletes a change
   rpc DeleteChange(DeleteChangeRequest) returns (DeleteChangeResponse);
+  // Lists all changes for a snapshot UUID
+  rpc ListChangesBySnaphotUUID(ListChangesBySnapshotUUIDRequest) returns (ListChangesBySnapshotUUIDResponse);
```
Comment (DavidS-ovm) [inline changes.proto:75]:
```suggestion
  rpc ListChangesBySnapshotUUID(ListChangesBySnapshotUUIDRequest) returns (ListChangesBySnapshotUUIDResponse);
```

File: gateway.proto [hunk 3]
```diff
@@ -205,3 +208,14 @@ message SnapshotLoadResult {
...
+
+message ChatMessage {
+    // The message to create
+    string message = 1;
+}
+
+message ChatMessageResult {
+    bool success = 1;
+    string errorMessage = 2;
+    string message = 3;
```
Comment (DavidS-ovm) [inline gateway.proto:220]: Assuming that `success` indicates which of errorMessage or message has a value, I'd suggest using 

```suggestion
    oneof Result {
        string message = 1;
        string error = 2;
    }
```

to ensure that there can only be one.

**Review by DavidS-ovm — APPROVED** [review]
Approving with two small comments.

File: gateway.proto (hunk 3 above)
Comment (tphoney) [reply gateway.proto]: i used oneof response_type, mimicing other gateway responses

**Pull request merged**